src/address.rs

```diff
@@ -4,74 +4,64 @@ use crate::utils::hex_str_to_bytes;
 use crate::utils::ArrayString;
 use bech32::{self, FromBase32};
 use bech32::{ToBase32, Variant};
+use core::fmt::Display;
 use std::fmt;
-use std::fmt::Display;
-use std::fmt::Formatter;
+use std::fmt::{Debug, Formatter};
 use std::str::FromStr;
 
-/// An address that's derived from a given PublicKey
+/// In cases where it's impossible to know the Bech32 prefix
+/// we fall back to this value
+pub const DEFAULT_PREFIX: &'static str = "cosmos";
+
+///
 #[derive(PartialEq, Eq, Copy, Clone, Hash, Deserialize, Serialize)]
-pub struct Address {
+pub enum Address {
+    Base(BaseAddress),
```

Comment:
can you doc comment what each time is? also incomplete doc comment for the enum itself.